packages/swingset-theme-hashicorp/src/components/live-component/index.tsx

```diff
@@ -0,0 +1,43 @@
+'use client'
+import {
+  SandpackProvider,
+  SandpackLayout,
+  SandpackCodeEditor,
+  SandpackPreview,
+} from '@codesandbox/sandpack-react'
+import { Children } from 'react'
+import { getFileMap } from './get-file-map'
+import { sandpackTheme } from './code-theme'
+
+type LiveComponentProps = {
+  children: string[]
+  deps?: Record<string, string>
+}
+
+export function LiveComponent({ children, deps }: LiveComponentProps) {
+  const codeSnippets = Children.toArray(children) as React.ReactElement[]
```

Comment:
Casts like `as React.ReactElement[]` are usually an indication that you're not using the correct types for what you're trying to do. What's the native return type of `Children.toArray()`?

Reply:
It's a union with several React component types `React.ReactNode | React.ReactNode[]): (string | number | React.ReactElement<any, string | React.JSXElementConstructor<any>> | React.ReactFragment | React.ReactPortal)[]` I casted to narrow down.